File: java/SpartacusSearch.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SpartacusSearch {

    public static <T> List<T> spartacusSearch(List<T> haystack, T needle) {
        List<T> results = new ArrayList<>(haystack.size());
        for (T x : haystack) {
            System.out.println(x);
            results.add(x);
        }
        return results;
    }

    public static void main(String[] args) {
        System.out.println(spartacusSearch(Arrays.asList(1, 2, 3, 4, 5), 3));
    }
}

File: javascript/spartacus-search.js
function spartacusSearch(haystack, needle) {
  if (!Array.isArray(haystack)) throw new TypeError('haystack must be an Array!');
  const results = [];
  for (const x of haystack) {
    console.log(x);
    results.push(x);
  }
  return results;
}

module.exports = spartacusSearch;

if (require.main === module) {
  console.log(spartacusSearch([1, 2, 3, 4, 5], 3));
  console.log(spartacusSearch(['a', 'b', 'c'], 'z'));
}
